radicle-cli/src/terminal/args.rs

```diff
@@ -99,6 +99,14 @@ pub fn rid(val: &OsString) -> anyhow::Result<Id> {
     Id::from_str(&val).map_err(|_| anyhow!("invalid Repository ID '{}'", val))
 }
 
+pub fn pubkey(val: &OsString) -> anyhow::Result<NodeId> {
+    let Ok(did) = did(val) else {
+        let nid = nid(val)?;
+        return Ok(nid)
+    };
+    Ok(did.as_key().to_owned())
```

Comment:
Pretty sure this can be cleaned up into:
```suggestion
    did(val).map(|did| *did.as_key()).or(nid(val))
```

Reply:
It feels subjective.  Its certainly more concise.  For me, ones clearer, the others shorter.  More people can understand the former quicker than the latter.

Reply:
I do not feel comfortable with this one-line solution sorry! My 2 sats areis to accept the verbosity when the code is not a complete mess. I am pretty sure that from your code style, you are a functional programming language guys, but I think the original solution is more debuggable.

It is also easier to read from people that are new to rust, and this mean be contribution friendly for people that came from Java or C. I will not apply this directly without a consensus between us on how the code should be written, and maybe put some rules somewhere

In rust the `?` can be a foot gun already when you are debugging, so if this code need to be debugged some days the single line that you suggest need to be rewritten in a more step-by-step solution

Reply:
> you are a functional programming language guys

:detective: 

That's all fair. It's not a hill I'm willing to die on :) If this way helps then, by all means, sounds good.